wagtail/admin/templates/wagtailadmin/shared/panel.html

```diff
@@ -19,23 +19,23 @@
 {% fragment as panel_id %}{{ prefix }}-section{% endfragment %}
 {% fragment as heading_id %}{{ prefix }}-heading{% endfragment %}
 {% fragment as content_id %}{{ prefix }}-content{% endfragment %}
-<section class="w-panel {{ classname }}" id="{{ panel_id }}" {% if heading %}aria-labelledby="{{ heading_id }}"{% endif %} data-panel>
+<section class="w-panel {{ classname }}" id="{{ panel_id }}" {% if heading %}aria-labelledby="{{ heading_id }}"{% endif %} w-panel data-controller="w-panel" >
     {# If a panel has no heading nor header controls, we don’t want any of the associated UI. #}
     {% if heading or header_controls %}
-        <div class="w-panel__header">
-            <a class="w-panel__anchor w-panel__anchor--prefix" href="#{{ panel_id }}" data-panel-anchor aria-labelledby="{{ heading_id }}">
+        <div class="w-panel__header" >
+            <a class="w-panel__anchor w-panel__anchor--prefix" href="#{{ panel_id }}" w-panel-anchor aria-labelledby="{{ heading_id }}">
                 {% icon name="link" classname="w-panel__icon" %}
             </a>
-            <button class="w-panel__toggle" type="button" aria-label="{% trans 'Toggle section' %}" aria-describedby="{{ heading_id }}" data-panel-toggle aria-controls="{{ content_id }}" aria-expanded="true">
+            <button class="w-panel__toggle" id="toggle-button" type="button" aria-label="{% trans 'Toggle section' %}" aria-describedby="{{ heading_id }}" data-w-panel-toggle data-action="click->w-panel#toggle" aria-controls="{{ content_id }}" aria-expanded="true">
```

Comment:
```suggestion
            <button class="w-panel__toggle" type="button" aria-label="{% trans 'Toggle section' %}" aria-describedby="{{ heading_id }}" data-action="w-panel#toggle" aria-controls="{{ content_id }}" aria-expanded="true">
```

Avoid setting an id on elements in this shared code, reminder that html must not have duplicate IDs. 

Similar to above, if we don't need the original panel data attributes, we should remove them. 

Finally, you can use the shorter method of declaring actions on some elements like the click event on a button.